KomaMRIFiles/src/Sequence/Pulseq.jl

```diff
@@ -390,12 +434,32 @@ function read_seq(filename)
             elseif  section == "[SHAPES]"
                 shapeLibrary = read_shapes(io, (version_major==1 && version_minor<4))
             elseif  section == "[EXTENSIONS]"
-                extensionLibrary = read_events(io,[1 1 1]) #For now, it reads the extensions but it does not take it them into account
-            elseif  section == "extension TRIGGERS 1"
-                triggerLibrary = read_events(io,[1 1 1e-6 1e-6])
+                extensionLibrary = read_extension_blocks(io)
             elseif  section == "[SIGNATURE]"
                 signature = read_signature(io)
             else
+                if startswith(section, "extension")
+                    extension = section[11:end]
+                    if startswith(extension, "TRIGGERS")
+                        id = parse(Int, extension[9:end])
+                        extensionType[id] = Dict("data"=>"TRIGGERS")
+                        triggerLibrary = read_events(io, [1, 1, 1e-6, 1e-6]; eventLibrary = triggerLibrary)
+                    elseif startswith(extension, "LABELSET")
+                        id = parse(Int, extension[9:end])
+                        extensionType[id] = Dict("data"=>"LABELSET")
+                        labelsetLibrary = read_labels(io;eventLibrary=labelsetLibrary)
+                    elseif startswith(extension, "LABELINC")
+                        id = parse(Int, extension[9:end])
+                        extensionType[id] = Dict("data"=>"LABELINC")
+                        labelincLibrary = read_labels(io; eventLibrary=labelincLibrary)
+                    elseif startswith(extension, "DELAYS")
+                        @warn "DELAYS extension is not handle"
+                    else
+                        @warn "Ignoring unknown extension, input string: $extension"
+                    end
```

Comment:
It would be nice to do this with multiple dispatch, as extensions are meant to be extended. So if the user creates a new extension called "MyEXT" they can just define maybe `read_extenstion(..., Val("MyEXT"))` or other functions like that, so they don't need to modify the code.